lib/spack/docs/known_issues.rst

```diff
@@ -77,6 +77,27 @@ See https://github.com/spack/spack/issues/267 and
 https://github.com/spack/spack/issues/2546 for further details.
 
 
+----------------------------------------------------
+Software stacks involving Python3 fail to concretize
+----------------------------------------------------
+
+This can happen for two reasons, both related to the concretizer:
+
+1. A package requires ``python@2`` as a build dependency only (eg:
```

Comment:
The inverse of this happens as well.  For instance, mesa required python 3 as a build-only dependency, but paraview requires python 2.  Thus `paraview+python` fails to concretize.  The workaround at the moment is to do a two stage install where you first install mesa explicitly.  Then when you install par5aview, the concretizer only evaluates the link and runtime dependencies of the installed mesa.